kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/HasMetadataOperation.java

```diff
@@ -42,18 +49,63 @@ public HasMetadataOperation(OperationContext ctx) {
   @Override
   public T edit(UnaryOperator<T> function) {
     T item = getMandatory();
-    return patch(function.apply(item));
+    T clone = Serialization.clone(item);
+    return patch(null, clone, function.apply(item), false);
+  }
+  
+  @Override
+  public T editStatus(UnaryOperator<T> function) {
+    T item = getMandatory();
+    T clone = Serialization.clone(item);
+    return patch(null, clone, function.apply(item), true);
   }
 
   @Override
   public T accept(Consumer<T> consumer) {
     T item = getMandatory();
+    T clone = Serialization.clone(item);
     consumer.accept(item);
-    return patch(item);
+    return patch(null, clone, item, false);
   }
-
+  
+  protected VisitableBuilder<T, ?> createVisitableBuilder(T item) {
+    throw new KubernetesClientException(NO_BUILDER);
+  }
+  
+  @Override
+  public T edit(Visitor... visitors) {
+    T item = getMandatory();
+    T clone = Serialization.clone(item);
+    return patch(null, clone, createVisitableBuilder(item).accept(visitors).build(), false);
+  }
+  
+  /**
+   * Get the current item from the server, consulting the metadata for the name if needed
+   */
+  protected Gettable<T> fromServer(ObjectMeta metadata) {
```

Comment:
why protected and not private?

Reply:
It can be whatever you want.  BaseOperation.fromServer is public, but I was unsure if this should be as well, so I went with protected.

Reply:
I really don't see how a children class would need to reimplement this, anyway, let's leave it and get this merged :rocket: